docs/dev/testing.rst

```diff
@@ -0,0 +1,306 @@
+section called ``commands`` is a list of shell commands that that test
+environment will run.
+
+-------------------------------------------------------------------------------
+Input Data for Integration Tests
+-------------------------------------------------------------------------------
+The software integration tests need a year's worth of input data to process. By
+default they will look in your local PUDL datastore to find it. If the data
+they need isn't available locally, they will download it from Zenodo and put it
+in the local datastore.
+
+However, if you're editing code that affects how the datastore works, you
+probably don't want to risk contaminating your working datastore. You can
+use a disposable temporary datastore instead by having Tox pass the
+``--tmp-data`` flag in to ``pytest`` like this:
+
+.. code-block:: console
+
+   $ tox -e integration -- --tmp-data
+
+The floating ``--`` isn't a typo, it tells Tox that you're done giving it
+command line arguments, and that any additional arguments it gets should be
+passed through to ``pytest``. We've configured ``pytest`` (through the
+``test/conftest.py`` configuration file) to be on the lookout for the
+``--tmp-data`` flag and act accordingly.
+
+.. seealso::
+
+    * :doc:`/usage` for more on how to set up a PUDL workspace, including a
+      datastore.
+    * :doc:`/datastore` for more on how to work with the datastore.
+
+-------------------------------------------------------------------------------
+Running pytest Directly
+-------------------------------------------------------------------------------
+Running tests directly with ``pytest`` gives you the ability to run only
+tests from a particular test module, or even a single individual test case.
+It's also faster because there's no testing environment to set up. Instead,
+it just uses your Python environment, which should be the ``pudl-dev`` conda
+environment discussed in :doc:`/dev/dev_setup`. This is convenient if you're
+debugging something specific, or developing new test cases, but it's not as
+robust as using Tox.
+
+Running specific tests
+^^^^^^^^^^^^^^^^^^^^^^
+To run the software unit tests with ``pytest`` directly:
+
+.. code-block:: console
+
+   $ pytest test/unit
+
+To run only the unit tests for the Excel spreadsheet extraction module:
+
+.. code-block:: console
+
+   $ pytest test/unit/extract/excel_test.py
+
+To run only the unit tests defined by a single test class within that module:
+
+.. code-block:: console
+
+   $ pytest test/unit/extract/excel_test.py::TestGenericExtractor
+
+Custom PUDL pytest flags
+^^^^^^^^^^^^^^^^^^^^^^^^
+We have defined several custom flags to control pytest's behavior when running
+the PUDL tests. You can always check to see what custom flags exist by running
+``pytest --help`` and looking at the ``custom options`` section:
+
+.. code-block:: console
+
+  custom options:
+  --live-dbs            Use existing PUDL/FERC1 DBs instead of creating temporary ones.
```

Comment:
so you can only specify live-ness in general - not individually?

Reply:
Yes. I was trying to keep it simple. The single-year ferc1 DB that includes only the tables which are integrated into PUDL that we use for testing only takes about 5 seconds to generate. Is there a scenario you have in mind where we want to use a pre-existing ferc1 DB, but want to create the pudl DB from scratch?

Reply:
running the full etl?

Reply:
This is what we have `pudl_etl` for right?

Reply:
you are probably right. i have definitely run the full etl as a test in the past, but that is by no means a default behavior.